### PR TITLE
[chore] parallelize tests on GPUs

### PR DESCRIPTION
**Pull request by min-xu-ai**
# Before submitting

- [ ] Was this discussed/approved via a Github issue? (no need for typos, doc improvements)
- [ ] Did you read the [contributor guideline](https://github.com/facebookresearch/fairscale/blob/master/CONTRIBUTING.md)?
- [ ] Did you make sure to update the docs?
- [ ] Did you write any new necessary tests?

## What does this PR do?
Fixes # (issue).

## PR review
Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

**Comment by min-xu-ai** [comment]
image


This PR parallelize the tests to make them go fast. As a nice side-effect, they don't affect each other too.

File: .circleci/config.yml [hunk 1]
```diff
@@ -5,7 +5,7 @@
 # Adopted from
 # https://github.com/facebookresearch/detectron2/blob/master/.circleci/config.yml
 
-version: 2
+version: 2.1
```
Comment (anj-s) [inline .circleci/config.yml:8]: Does the workflows version also need to change?

File: .circleci/config.yml [hunk 2]
```diff
@@ -141,6 +141,10 @@ run_flake8: &run_flake8
         flake8 --show-source --statistics
 
 
+# TODO (Min): figure out how to do coverage nightly or on-demand. Doing it
```
Comment (anj-s) [inline .circleci/config.yml:144]: +1 I did a bit of digging regarding this last month. We can definitely do nightly coverage. The first step is to figure out which tests should be nightly and which are per commit.

**Review by anj-s — APPROVED** [review]

File: .circleci/config.yml (hunk 1 above)
Comment (min-xu-ai) [reply .circleci/config.yml:8]: yeah, it seems to be needed due to parameterization in the config.

**Comment by min-xu-ai** [comment]
Thanks @anj-s

**Pull request merged**